src/schoology-extractor/schoology_extractor/helpers/sync.py

```diff
@@ -40,21 +41,30 @@ def _get_current_date_with_format() -> str:
     return datetime.strftime(datetime.now(), "%Y-%m-%d %H:%M:%S")
 
 
+def _map_result_to_dicts_arr(result: Union[list[ResultProxy], None]):
+    mapped_arr: list[dict] = list[dict]()
+    if result is None:
+        return list()
+    for row in result:
+        db_item = dict()
+        for column, db_value in row.items():  # type: ignore
+            db_item[column] = db_value
+        mapped_arr.append(db_item)
```

Comment:
You might try something like this instead:

```
mapped_arr = [dict(row.items()) for row in result]
```

Reply:
Your code seems fine by the way. Just might be a small improvement, especially with larger results, to use list comprehension instead and take better advantage of SQL Alchemy.

Reply:
✔️